### PR TITLE
[Major version bump] Dragon slayer improvements

### PR DESCRIPTION
**Pull request by jstefaniuk-d2l**
Upgrade to d2l-image-tile, hybridize the bower dependencies, clean up linting and the demo page

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -1,33 +1,55 @@
...
     "d2l-hm-constants-behavior": "git://github.com/Brightspace/d2l-hm-constants-behavior.git#^3.0.0",
     "d2l-icons": "^3.0.0",
     "d2l-organization-hm-behavior": "git://github.com/Brightspace/organization-hm-behavior.git#^1.0.0",
-    "d2l-tile": "git://github.com/Brightspace/d2l-tile#^1.0.2",
+    "d2l-tile": "^2.0.1",
     "d2l-image": "git://github.com/Brightspace/d2l-image#^1.1.0",
     "d2l-user-profile-behavior": "git://github.com/Brightspace/user-profile-behavior#^2.1.0",
-    "polymer": "^1.8.0",
-    "d2l-colors": "^2.3.0"
+    "d2l-colors": "^2.4.0 || ^3.1.0",
```
Comment (ChrisKraljevicD2L) [inline bower.json]: Nit: Should these deps be alphabetized?

File: d2l-user-tile-auto.html [hunk 2]
```diff
@@ -45,7 +48,12 @@
 				type: Object,
 				value: null
 			},
-			_doneRequests: Boolean
+			_doneRequests: Boolean,
+			_name: String,
```
Comment (ryantmer) [inline d2l-user-tile-auto.html]: Huh, surprised these weren't causing polymer lint failures before, it generally complains about using bindings that aren't defined properties.

File: d2l-user-tile-auto.html [hunk 3]
```diff
@@ -4,6 +4,9 @@
 <link rel="import" href="../d2l-user-profile-behavior/d2l-user-profile-behavior.html">
 <link rel="import" href="d2l-user-tile.html">
 
+<!--
+`d2l-user-tile-auto` is a Polymer-based web component for creating a user tile which automatically fills itself out when given a token.
```
Comment (ryantmer) [inline d2l-user-tile-auto.html]: When given a token and URL, strictly speaking

File: d2l-user-tile.html [hunk 4]
```diff
@@ -90,7 +93,13 @@
 			}
 		</style>
 
-		<d2l-tile aria-label$="[[name]]" img-url="[[_getImgUrl(background, _placeholders)]]" style$="[[_createBackgroundStyle(backgroundColor, _placeholders)]]" hover-effect="low-lift" loading$="[[placeholders]]">
+		<d2l-image-tile
+			aria-label$="[[name]]"
+			img-url="[[_getImgUrl(background, _placeholders)]]"
```
Comment (ryantmer) [inline d2l-user-tile.html]: Looks like `background` and `backgroundColor` are missing the property definitions

**Comment by jstefaniuk-d2l** [comment]
Breaking change is in the tile dependency

File: d2l-user-tile.html (hunk 4 above)
Comment (jstefaniuk-d2l) [reply d2l-user-tile.html]: Nope: https://github.com/Brightspace/user-tile/pull/45/files/542dfb3b6bccebf35819c112308b3c596c5b0797#diff-c1a2647e4679e9ed3c7eb3a0ace0fd20R126

File: d2l-user-tile-auto.html (hunk 2 above)
Comment (jstefaniuk-d2l) [reply d2l-user-tile-auto.html]: I think the old linting file was less than great

File: d2l-user-tile.html (hunk 4 above)
Comment (ryantmer) [reply d2l-user-tile.html]: My reading comprehension 0/10 today apparently 😛

**Pull request merged**